File: 0599-minimum-index-sum-of-two-lists/0599-minimum-index-sum-of-two-lists.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        min_i_sum = float("inf")
        min_s = [""]
        for i in list1:
            if i in list2:
                if(list1.index(i) + list2.index(i)) < min_i_sum:
                    min_i_sum = list1.index(i) + list2.index(i)
                    min_s[-1] = i
                elif (list1.index(i) + list2.index(i)) == min_i_sum:
                    min_s.append(i)
        return min_s

        

File: 0599-minimum-index-sum-of-two-lists/0599-minimum-index-sum-of-two-lists.go
func findRestaurant(list1 []string, list2 []string) []string {
	restMap := make(map[string]int)
	for i, rs := range list1 {
		restMap[rs] = i
	}

	minIndexSum := math.MaxInt32
	var res []string
	for i, rs := range list2 {
		if index, found := restMap[rs]; found {
			indexSum := i + index
			if indexSum < minIndexSum {
				minIndexSum = indexSum
				res = []string{rs}
			} else if indexSum == minIndexSum {
				res = append(res, rs)
			}
		}
	}
	return res
}